lib/xebow_web/live/matrix_live.html.leex

```diff
@@ -0,0 +1,34 @@
+<div class="row">
+  <div class="column column-10">
+    <button class="button button-clear">Animations:</button>
+  </div>
+  <div class="column column-90">
+    <button phx-click="previous_animation">← previous</button>
+    <button phx-click="next_animation">next →</button>
+  </div>
+</div>
+
+<hr>
+
+<div class="screen">
+  <%= for led <- @leds do %>
+    <span
+      class="<%= led.class %>"
+      <%= if led.class == "key" do %>
+      phx-click="key_pressed"
+      phx-value-key-id="<%= led.id %>"
+      <% end %>
+      style="
+        left: <%= led.x %>px;
+        top: <%= led.y %>px;
+        width: <%= led.width %>px;
+        height: <%= led.height %>px;
+        background-color: <%= led.color %>
+      "
+    ></span>
+  <% end %>
```

Comment:
Nice debugging so far!

The biggest question in my mind is if LiveView is only supposed to be sending the properties that have changed when an update is performed, why does it not seem to be doing it in this case? If the property changing during an animation is `background-color`, why is a bunch of other data being sent down the wire when the LED colors are updated? For updating 12 LEDs each with a 6 character hex color value and serialized to JSON, with a little extra overhead in there, I would expect 100-200 bytes to be in the ballpark. We've probably done something unintentionally do deoptimize LiveView's diffing. I don't see anything in [the docs](https://hexdocs.pm/phoenix_live_view/0.13.3/Phoenix.LiveView.html#module-assigns-and-liveeex-templates) that stands out yet.

>I wonder if there's a way to make each display tell the engine how frequently it wants updates, and/or reduce the amount of data transferred in each diff.

If we can't figure out how to optimize our LiveView code, then we could debounce/drop frames in the LiveView process. This could either be done in `handle_info`, in which case the LiveView process will still have the overhead of receiving the messages in its mailbox, or in the callback function, which could drop frames before `send` is called.